src/features/like/like-slice.js

```diff
@@ -0,0 +1,75 @@
+import { createSlice, createAsyncThunk } from "@reduxjs/toolkit";
+import axios from "axios";
+
+const encodedToken = localStorage.getItem("token");
```

Comment:
Here you are getting time. Outside function this will cause problem 
You have to reload page after login to work.
Bcz this slice will render only once and that time u don't have token in local storage
Yet, and after login when u have token slice won't be render.
By refreshing u r forcing render that's why It is working after refresh
To prevent this call this getToken inside liked page or any other page pass it with dispatch

Reply:
Thank you, you helped me solve a big problem. Took your suggestion, implemented it and now it is working fine.